litmus-portal/graphql-server/pkg/handlers/file_handler.go

```diff
@@ -53,45 +60,48 @@ func FileHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func GetManifest(token string) ([]byte, int, error) {
-
-	id, err := cluster.ClusterValidateJWT(token)
+	clusterID, err := cluster.ClusterValidateJWT(token)
 	if err != nil {
 		return nil, 404, err
 	}
 
-	reqCluster, err := dbOperationsCluster.GetCluster(id)
+	reqCluster, err := dbOperationsCluster.GetCluster(clusterID)
 	if err != nil {
 		return nil, 500, err
 	}
 
-	scope := os.Getenv("PORTAL_SCOPE")
-	if scope == clusterScope {
+	if scope == namespaceScope && endpoint == "" {
+		return nil, 500, errors.New("CHAOS_CENTER_UI_ENDPOINT env is empty for namespace scope installation")
```

Comment:
@gdsoumya Self agent will work as it is..

1. If `controlplane scope is namespace` then, `CHAOS_CENTER_UI_ENDPOINT is required`. (that check wasn't present earlier)
2. If `controlplane scope is cluster and CHAOS_CENTER_UI_ENDPOINT is not present`, then `it will retrieve the endpoint from the cluster(existing logic)`

So, as part of some discussions, we are making CHAOS_CENTER_UI_ENDPOINT available for both scopes.

Reply:
I understand that but how will the user know beforehand what their chaos center UI endpoint is ? If they are running nodeport or just port-forward it's difficult to determine at installation time.  The only option the user will have in such case is to deploy the chaos center and then figure the ip/url and re-apply with changed config because even if they just want to playaround with self-agent and no external agent in ns scope the self-agent installation will fail without the UI endpoint en var set.

Reply:
for namespace mode- we already set the CHAOS_CENTER_UI_ENDPOINT to http://litmusportal-frontend-service:9091 as default

Reply:
for cluster scope only we are not setting it.